### DISCUSSION TITLE
Expected behavior when already logged in, and going to /create-account/

### DISCUSSION THREAD

**Discussion by winstonwolff** [jeremyevans/rodauth#312] [General]
Hi Jeremy—

In my testing, I run into errors when I'm logged in already, and then I go to /create-account/. I get the error "Validation failed: Email has already been taken"

We're probably going to fix it by logging the user out when they go to /create-account/. But I wonder if that is expected behavior for a good reason? Would Rodauth want to handle that case internally?

—Winston

**Comment by jeremyevans** [comment]
You can use the `already_logged_in` configuration method to override how to handle cases when logged in users go to a page that generally is used when not logged in.  A common use here would be to set a flash notice/error and redirect to another page.  By default, the hook doesn't take any action.

Not sure why you are getting the error you are seeing, unless you are trying to use an email that does already exist.  At least from some simple testing in the specs, logged in users should be able to create accounts for email addresses that don't already have accounts.

**Comment by winstonwolff** [comment]
So I think what's happening, is during our testing, we create a lot of accounts. If we start to create an account, but haven't finished the required second factor, one has a half-built account. If you try to go to /create-account again, it gets the error above. For anybody else who happens upon this situation, here's what we did to avoid this issue:

```
before_create_account_route do
  logout
end
```
